### PR TITLE
Add `Regexp.linear_time?`

### PR DESCRIPTION
**Pull request by makenowjust**
Add a new method `Regexp.linear_time?` to check if matching against a given regexp can be completed in linear time by the optimization introduced in #6486.

### CODE REVIEW

**Comment by makenowjust** [comment]
CI has fault on MinGW (UCRT64). I'm not sure this failure is related to this PR. Should I re-run this job?

**Comment by hsbt** [comment]
> Should I re-run this job?

👍  I re-run it. But MinGW environment sometimes failure with unrelated changes. We can merge PR without passing MinGW if it randomly failed again and again.

File: re.c [hunk 1]
```diff
@@ -4194,6 +4194,39 @@ rb_reg_s_union_m(VALUE self, VALUE args)
...
 
+/*
+ *  call-seq:
+ *    Regexp.linear_time?(re)
+ *    Regexp.linear_time?(string, options = 0)
+ *
+ *  Returns +true+ if matching against <tt>re</tt> can be
+ *  done in linear time to the input string.
+ *
+ *    Regexp.linear_time?(/re/) # => true
```
Comment (makenowjust) [inline re.c:4205]: Please tell me if this documentation is sufficient.

Comment (mame) [reply re.c:4205]: I think it's good enough.

**Pull request merged**